### PR TITLE
Fix lint errors

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: cocos-creator/3d-tasks#

Changelog:
 * Fix some lint errors.

This is a prerequisite of https://github.com/cocos-creator/3d-tasks/issues/5419 . Request review from @jareguo 

Changes included are mostly emitted by eslint auto-fix.

### CODE REVIEW

File: cocos/core/scene-graph/base-node.ts [hunk 1]
```diff
@@ -222,32 +218,28 @@ export class BaseNode extends CCObject implements ISchedulable {
...
     protected static _stacks: Array<Array<(BaseNode | null)>> = [[]];
     protected static _stackId = 0;
 
-    protected static _findComponent (node: BaseNode, constructor: Function) {
+    /**
+     * Call `_updateScene` of specified node.
+     * @param node The node.
+     */
+    protected static _setScene (node: BaseNode) {
+        node._updateScene();
```
Comment (shrinktofit) [inline cocos/core/scene-graph/base-node.ts:230]: I do think this is better than `if (node instanceof ...)`. How do you think? @jareguo

File: cocos/core/scene-graph/base-node.ts [hunk 2]
```diff
@@ -336,7 +328,19 @@ export class BaseNode extends CCObject implements ISchedulable {
 
     protected _siblingIndex = 0;
 
-    protected _registerIfAttached = !EDITOR ? undefined : function (this: BaseNode, register) {
+    /**
+     * Set `_scene` field of this node.
+     * The derived `Scene` overrides this method to behavior differently.
+     */
+    protected _updateScene () {
+        if (this._parent == null) {
```
Comment (shrinktofit) [inline cocos/core/scene-graph/base-node.ts:336]: As doc says, `Scene._udpateScene` overrides this implementation. Then we eliminate a potential circular reference.

File: cocos/core/scene-graph/base-node.ts [hunk 3]
```diff
@@ -40,17 +41,13 @@ import IdGenerator from '../utils/id-generator';
...
 import { Node } from './node';
-import { Scene } from './scene';
-
-type Constructor<T = {}> = new (...args: any[]) => T;
+import type { Scene } from './scene';
 
 const Destroying = CCObject.Flags.Destroying;
 const DontDestroy = CCObject.Flags.DontDestroy;
 const Deactivating = CCObject.Flags.Deactivating;
-// @ts-expect-error
```
Comment (shrinktofit) [inline cocos/core/scene-graph/base-node.ts:54]: I have no idea about this error. @jareguo

File: cocos/core/scene-graph/node-event-processor.ts [hunk 4]
```diff
@@ -150,11 +150,11 @@ function _mouseMoveHandler (this: EventListener, event: EventMouse) {
     if (hit) {
```
Comment (shrinktofit) [inline cocos/core/scene-graph/node-event-processor.ts:150]: Do you have any thought on the above lint error? @YunHsiao

File: cocos/core/scene-graph/base-node.ts (hunk 1 above)
Comment (jareguo) [reply cocos/core/scene-graph/base-node.ts:230]: Agree

File: cocos/core/scene-graph/base-node.ts (hunk 2 above)
Comment (jareguo) [reply cocos/core/scene-graph/base-node.ts:335]: Or we can just set `scene._scene = scene;`

**Review by jareguo — APPROVED** [review]

**Pull request merged**